backend/src/main/kotlin/org/loculus/backend/db/migration/V1_18_1__Migrate_Sequence_Entries_To_Use_Compression_Dict.kt

```diff
@@ -0,0 +1,229 @@
+@file:Suppress("ktlint:standard:filename", "ktlint:standard:class-naming")
+
+package org.loculus.backend.db.migration
+
+import mu.KotlinLogging
+import org.flywaydb.core.api.migration.BaseJavaMigration
+import org.flywaydb.core.api.migration.Context
+import org.jetbrains.exposed.sql.Database
+import org.jetbrains.exposed.sql.JoinType
+import org.jetbrains.exposed.sql.SqlExpressionBuilder.eq
+import org.jetbrains.exposed.sql.Table
+import org.jetbrains.exposed.sql.and
+import org.jetbrains.exposed.sql.transactions.transaction
+import org.jetbrains.exposed.sql.update
+import org.loculus.backend.api.Organism
+import org.loculus.backend.api.OriginalData
+import org.loculus.backend.api.ProcessedData
+import org.loculus.backend.service.jacksonSerializableJsonb
+import org.loculus.backend.service.submission.CompressionDictService
+import org.loculus.backend.utils.Accession
+import org.loculus.backend.utils.Version
+import org.springframework.stereotype.Component
+
+private val log = KotlinLogging.logger { }
+
+@Component
+class V1_18_1__Migrate_Sequence_Entries_To_Use_Compression_Dict(
+    private val compressionDictService: CompressionDictService,
+) : BaseJavaMigration() {
+
+    override fun migrate(context: Context) {
+        log.info { "Starting migration" }
+
+        val db = Database.connect(context.configuration.dataSource)
+
+        var i = 0
+
+        transaction(db) {
+            V1_18_1_SequenceEntriesTable
+                .select(
+                    V1_18_1_SequenceEntriesTable.accessionColumn,
+                    V1_18_1_SequenceEntriesTable.versionColumn,
+                    V1_18_1_SequenceEntriesTable.organismColumn,
+                    V1_18_1_SequenceEntriesTable.originalDataColumn,
+                )
+                .fetchSize(1000)
+                .forEach { row ->
+                    val accession = row[V1_18_1_SequenceEntriesTable.accessionColumn]
+                    val version = row[V1_18_1_SequenceEntriesTable.versionColumn]
+                    val organism = row[V1_18_1_SequenceEntriesTable.organismColumn]
+                    val originalData = row[V1_18_1_SequenceEntriesTable.originalDataColumn] ?: return@forEach
+
+                    val migratedOriginalData = OriginalData(
+                        metadata = originalData.metadata,
+                        files = originalData.files,
+                        unalignedNucleotideSequences = originalData.unalignedNucleotideSequences.mapValues {
+                            val value = it.value
+                            when {
+                                value == null -> null
+                                value.compressionDictId !== null -> value
+                                else -> CompressedSequence(
+                                    compressedSequence = value.compressedSequence,
+                                    compressionDictId = compressionDictService
```

Comment:
@fengelniederhammer sorry if this is stupid but Im now rereading and I am confused - are't all the old unaligned sequences compressed on a segment level? Don't we need to get the dict from `getDictForSegmentOrGene`?

Reply:
Reading this now, I think you're right.